components/odh-notebook-controller/e2e/notebook_controller_setup_test.go

```diff
@@ -0,0 +1,92 @@
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	nbv1 "github.com/kubeflow/kubeflow/components/notebook-controller/api/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	k8sclient "k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlruntime "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var (
+	resourceCreationTimeout = time.Minute * 1
+	resourceRetryInterval   = time.Second * 10
+	notebookTestNamespace   string
+	skipDeletion            bool
+	scheme                  = runtime.NewScheme()
+)
+
+// Holds information specific to individual tests
+type testContext struct {
+	// Rest config
+	cfg *rest.Config
+	// client for k8s resources
+	kubeClient *k8sclient.Clientset
+	// custom client for managing cutom resources
+	customClient client.Client
+	// namespace for running the tests
+	testNamespace string
+}
+
+func NewTestContext() (*testContext, error) {
+
+	// GetConfig(): If KUBECONFIG env variable is set, it is used to create
+	// the client, else the inClusterConfig() is used.
+	// Lastly if none of the them are set, it uses  $HOME/.kube/config to create the client.
+	config, err := ctrlruntime.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error creating the config object %v", err)
+	}
+
+	kc, err := k8sclient.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to initialize Kubernetes client")
+	}
+
+	// custom client to manages resources like Notebook, Route etc
+	custClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to initialize custom client")
+	}
+
+	return &testContext{
+		cfg:           config,
+		kubeClient:    kc,
+		customClient:  custClient,
+		testNamespace: notebookTestNamespace,
+	}, nil
+}
+
+// TestKFNBC sets up the testing suite for KFNBC.
+func TestKFNBC(t *testing.T) {
```

Comment:
Rename `TestKFNBC` with something like `TestE2ENotebookController` since KFNBC refers only to Kubeflow version and we are testing the ODH extension too